hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationManager.java

```diff
@@ -185,9 +184,8 @@
                 executionService, migrationPauseDelayMs, 2 * migrationPauseDelayMs, this::resumeMigration);
         this.memberHeartbeatTimeoutMillis = properties.getMillis(ClusterProperty.MAX_NO_HEARTBEAT_SECONDS);
         nodeEngine.getMetricsRegistry().registerStaticMetrics(stats, PARTITIONS_PREFIX);
-        this.autoRebalanceDelaySeconds = properties.getInteger(PARTITION_REBALANCE_AFTER_MEMBER_LEFT_DELAY_SECONDS);
-        checkNotNegative(autoRebalanceDelaySeconds,
-                PARTITION_REBALANCE_AFTER_MEMBER_LEFT_DELAY_SECONDS.getName() + " must be not negative");
+        this.autoRebalanceDelaySeconds = node.getConfig().getPersistenceConfig().getRebalanceDelaySeconds();
+        checkNotNegative(autoRebalanceDelaySeconds, "Rebalance delay must be not negative");
```

Comment:
do we need this `checkNotNegative` check in here?

Reply:
You are right, it is redundant. Also, since rebalance delay only makes sense when persistence is enabled, in a further commit I added a check so the value is only taken into account with persistence.